### PR TITLE
Reconnecting websocket

### LINKED ISSUE #4: Reconnecting client to server
In case of a lost connection, the clients don't automatically reconnect. This would be desirable (although I can imagine that this is more a development problem than a problem in practice where the server should always be up-and-running.

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
Fixes #4

### CODE REVIEW

File: refbox/src/refBox.js [hunk 1]
```diff
@@ -176,34 +177,20 @@ class RefBox extends React.Component {
...
+    // listen to data sent from the websocket server
+    const message = JSON.parse(evt);
+    console.log('Message: ', message);
+    this.setState({dataFromServer: message});
+    if (this.state.arena in message) {
+      this.updateArenaData(message[this.state.arena]);
     }
+  }
 
+  componentDidMount() {
```
Comment (jlunenburg) [inline refbox/src/refBox.js]: This is deprecated isn't it?

File: refbox/src/audienceInfo.js [hunk 2]
```diff
@@ -167,38 +169,24 @@ class AudienceInfo extends React.Component {
...
-
-    this.ws.onclose = () => {
-      console.log('disconnected')
-      // automatically try to reconnect on connection loss
+    if ('metadata' in message && this.state.arena in message.metadata) {
+      this.updateMetaData(message.metadata[this.state.arena]);
     }
+  }
 
+  componentDidMount() {
```
Comment (jlunenburg) [inline refbox/src/audienceInfo.js]: Deprecated?

**Review by jlunenburg — DISMISSED** [review]
This looks good. I guess you tested it?

File: refbox/src/refBox.js (hunk 1 above)
Comment (MatthijsBurgh) [reply refbox/src/refBox.js]: Yes, I don't know react requires this function to be defined.

File: refbox/src/audienceInfo.js (hunk 2 above)
Comment (MatthijsBurgh) [reply refbox/src/audienceInfo.js]: Yes, I don't know react requires this function to be defined.

**Comment by MatthijsBurgh** [comment]
> This looks good. I guess you tested it?

Yes it reconnects. Though you should look into the behaviour of both apps on reconnecting.

File: refbox/src/refBox.js (hunk 1 above)
Comment (jlunenburg) [reply refbox/src/refBox.js]: Also works without, removed

**Review by jlunenburg — APPROVED** [review]

**Pull request merged**